src/test/java/lambdaperson/FilteredPersonsTest.java

```diff
@@ -0,0 +1,100 @@
+package lambdaperson;
+
+import org.junit.jupiter.api.Test;
+import personlist.ListActions;
+import personlist.Person;
+
+import java.util.Collections;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class FilteredPersonsTest {
+
+    @Test
+    void filteredSet() {
+        Set<Person> given = Set.of(
+                new Person("Stephen", 25),
+                new Person("John", 25),
+                new Person("John", 25),
+                new Person("Rose", 24),
+                new Person("Stephen", 16),
+                new Person("Stephen", 10),
+                new Person("John", 30)
+        );
+        Set<Person> expected = Set.of(
+                new Person("JOHN", 25),
+                new Person("ROSE", 24),
+                new Person("JOHN", 30),
```

Comment:
I'm not sure if this test succeeds since there's a condition: `Person.getAge() < 30`